db/migrate/20230313150008_create_solidus_stripe_customers.rb

```diff
@@ -0,0 +1,14 @@
+class CreateSolidusStripeCustomers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :solidus_stripe_customers do |t|
+      t.references :payment_method, null: false, foreign_key: { to_table: :spree_payment_methods }
+
+      t.string :source_type
+      t.integer :source_id
```

Comment:
I would add a multi column index and for added safety make it unique.
```
add_index : solidus_stripe_customers, [: payment_method, : source_type, :source_id], unique: true
```

I can't remember if the `t.references :payment_method` creates automatically the index but it's not performant enough especially if the query you will do most of the time is the one in the customer model, generated by the line
```
 find_or_initialize_by(payment_method: payment_method, source: order.user || order)
```

It would be much more efficient to find existing ones if you add this index.

Reply:
Also current code would not use it but still, for debugging or finding in console, I would also make a index on stripe_id column but that's more optionnal

Reply:
It'll make writes more expensive, but I agree it's better to default to the safe side

Reply:
Thanks for the feedback everyone, I'll add the index shortly, definitely an oversight not adding it 🙌